### PR TITLE
SW-1163 Remove Update and Delete Permission from Contributor

### PR DESCRIPTION
**Pull request by charlie-tf**
- Modify the canUpdateAccession method to deny for Contributor
- Note that canDeleteAccession is the same as canUpdateAccession
- Modify PermissionTest to reflect new permissions

### CODE REVIEW

File: src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt [hunk 1]
```diff
@@ -520,7 +520,7 @@ internal class PermissionTest : DatabaseTest() {
   }
 
   @Test
-  fun `contributors have access to data associated with their organizations`() {
+  fun `contributors have access to data associated with their organizations, but cannot update or delete accessions`() {
```
Comment (sgrimm) [inline src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt]: This feels too specific to me -- when we make a similar change for nursery support, we'd kind of need to update this to also describe the nursery-related limitations.

How about something like, `contributors have full read and limited write access to data associated with their organizations`?

File: src/main/kotlin/com/terraformation/backend/customer/model/IndividualUser.kt [hunk 2]
```diff
@@ -365,9 +365,15 @@ data class IndividualUser(
...
-    return canReadAccession(accessionId)
+    // Updating accession is allowed only for Role.MANAGER and higher.
+    // The user must also have read permissions for the accession.
+    val facilityId = parentStore.getFacilityId(accessionId)
+    return when (facilityRoles[facilityId]) {
+      Role.OWNER,
+      Role.ADMIN,
+      Role.MANAGER -> true
+      else -> false
+    } && canReadAccession(accessionId)
```
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/customer/model/IndividualUser.kt]: We don't need the read permission check here; it'll cause an extra `parentStore` data fetch on the happy path and will never be false when the `when` expression is true. The test suite should ensure the return value is always what we want.

**Review by sgrimm — CHANGES_REQUESTED** [review]
This is the right idea.

File: src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt (hunk 1 above)
Comment (charlie-tf) [reply src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt]: Yeah, that sounds like good verbiage. I'll go with it!

**Review by sgrimm — APPROVED** [review]
👍

**Pull request merged**